docker/test-images/zipkin-mysql/install.sh

```diff
@@ -15,6 +15,10 @@
 
 set -eux
 
+arch=${1#*/}
+if [ "$arch" = "s390x" ]; then
+    sed -i 's/\(.*\)ROW_FORMAT=COMPRESSED \(.*\)/\1\2/g' zipkin-schemas/mysql.sql
```

Comment:
Very small nit: Do you mind adding a comment why we need this?

Reply:
yeah, I got the following errors during the build for `s390x` (bu adding the docker buildx option `--progress plain`: 
```
#14 17.15 + mysql --verbose '--user=mysql' '--protocol=socket' -uroot
#14 17.24 --------------
#14 17.29 ERROR 4047 (HY000) at line 15 in file: 'zipkin-schemas/mysql.sql': InnoDB refuses to write tables with ROW_FORMAT=COMPRESSED or KEY_BLOCK_SIZE.
#14 17.29 ERROR 1146 (42S02) at line 28 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_spans' doesn't exist
#14 17.29 DELETE FROM mysql.user
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 DROP DATABASE IF EXISTS test
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 CREATE DATABASE zipkin
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 CREATE TABLE IF NOT EXISTS zipkin_spans (
#14 17.29   `trace_id_high` BIGINT NOT NULL DEFAULT 0 COMMENT 'If non zero, this means the trace uses 128 bit traceIds instead of 64 bit',
#14 17.29   `trace_id` BIGINT NOT NULL,
#14 17.29   `id` BIGINT NOT NULL,
#14 17.29   `name` VARCHAR(255) NOT NULL,
#14 17.29   `remote_service_name` VARCHAR(255),
#14 17.29   `parent_id` BIGINT,
#14 17.29   `debug` BIT(1),
#14 17.29   `start_ts` BIGINT COMMENT 'Span.timestamp(): epoch micros used for endTs query and to implement TTL',
#14 17.29   `duration` BIGINT COMMENT 'Span.duration(): micros used for minDuration and maxDuration query',
#14 17.29   PRIMARY KEY (`trace_id_high`, `trace_id`, `id`)
#14 17.29 ) ENGINE=InnoDB ROW_FORMAT=COMPRESSED CHARACTER SET=utf8 COLLATE utf8_general_ci
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 ALTER TABLE zipkin_spans ADD INDEX(`trace_id_high`, `trace_id`) COMMENT 'for getTracesByIds'
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 ALTER TABLE zipkin_spans ADD INDEX(`name`) COMMENT 'for getTraces and getSpanNames'ERROR 1146 (42S02) at line 29 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_spans' doesn't exist
#14 17.29 ERROR 1146 (42S02) at line 30 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_spans' doesn't exist
#14 17.29 ERROR 1146 (42S02) at line 31 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_spans' doesn't exist
#14 17.29 
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 ALTER TABLE zipkin_spans ADD INDEX(`remote_service_name`) COMMENT 'for getTraces and getRemoteServiceNames'
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 ALTER TABLE zipkin_spans ADD INDEX(`start_ts`) COMMENT 'for getTraces ordering and range'
#14 17.29 --------------
#14 17.29 
#14 17.29 --------------
#14 17.29 CREATE TABLE IF NOT EXISTS zipkin_annotations (
#14 17.29   `trace_id_high` BIGINT NOT NULL DEFAULT 0 COMMENT 'If non zero, this means the trace uses 128 bit traceIds instead of 64 bit',
#14 17.29   `trace_id` BIGINT NOT NULL COMMENT 'coincides with zipkin_spans.trace_id',
#14 17.29   `span_id` BIGINT NOT NULL COMMENT 'coincides with zipkin_spans.id',
#14 17.29   `a_key` VARCHAR(255) NOT NULL COMMENT 'BinaryAnnotation.key or Annotation.value if type == -1',
#14 17.29   `a_value` BLOB COMMENT 'BinaryAnnotation.value(), which must be smaller than 64KB',
#14 17.29   `a_type` INT NOT NULL COMMENT 'BinaryAnnotation.type() or -1 if Annotation',
#14 17.29   `a_timestamp` BIGINT COMMENT 'Used to implement TTL; Annotation.timestamp or zipkin_spans.timestamp',
#14 17.29   `endpoint_ipv4` INT COMMENT 'Null when Binary/Annotation.endpoint is null',
#14 17.29   `endpoint_ipv6` BINARY(16) COMMENT 'Null when Binary/Annotation.endpoint is null, or no IPv6 address',
#14 17.29   `endpoint_port` SMALLINT COMMENT 'Null when Binary/Annotation.endpoint is null',
#14 17.29   `endpoint_service_name` VARCHAR(255) COMMENT 'Null when Binary/Annotation.endpoint is null'
#14 17.29 ) ENGINE=InnoDB ROW_FORMAT=COMPRESSED CHARACTER SET=utf8 COLLATE utf8_general_ciERROR 4047 (HY000) at line 33 in file: 'zipkin-schemas/mysql.sql': InnoDB refuses to write tables with ROW_FORMAT=COMPRESSED or KEY_BLOCK_SIZE.
#14 17.30 ERROR 1146 (42S02) at line 47 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 48 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 49 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 50 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 51 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 52 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 ERROR 1146 (42S02) at line 53 in file: 'zipkin-schemas/mysql.sql': Table 'zipkin.zipkin_annotations' doesn't exist
#14 17.30 
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD UNIQUE KEY(`trace_id_high`, `trace_id`, `span_id`, `a_key`, `a_timestamp`) COMMENT 'Ignore insert on duplicate'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id_high`, `trace_id`, `span_id`) COMMENT 'for joining with zipkin_spans'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id_high`, `trace_id`) COMMENT 'for getTraces/ByIds'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`endpoint_service_name`) COMMENT 'for getTraces and getServiceNames'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`a_type`) COMMENT 'for getTraces and autocomplete values'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`a_key`) COMMENT 'for getTraces and autocomplete values'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------
#14 17.30 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id`, `span_id`, `a_key`) COMMENT 'for dependencies job'
#14 17.30 --------------
#14 17.30 
#14 17.30 --------------ERROR 4047 (HY000) at line 55 in file: 'zipkin-schemas/mysql.sql': InnoDB refuses to write tables with ROW_FORMAT=COMPRESSED or KEY_BLOCK_SIZE.
#14 17.32 
#14 17.32 CREATE TABLE IF NOT EXISTS zipkin_dependencies (
#14 17.32   `day` DATE NOT NULL,
#14 17.32   `parent` VARCHAR(255) NOT NULL,
#14 17.32   `child` VARCHAR(255) NOT NULL,
#14 17.32   `call_count` BIGINT,
#14 17.32   `error_count` BIGINT,
#14 17.32   PRIMARY KEY (`day`, `parent`, `child`)
#14 17.32 ) ENGINE=InnoDB ROW_FORMAT=COMPRESSED CHARACTER SET=utf8 COLLATE utf8_general_ci
#14 17.32 --------------
#14 17.32 
#14 17.32 --------------
#14 17.32 GRANT ALL PRIVILEGES ON zipkin.* TO zipkin@'%' IDENTIFIED BY 'zipkin' WITH GRANT OPTION
#14 17.32 --------------
#14 17.32 
#14 17.32 --------------
#14 17.32 FLUSH PRIVILEGES
#14 17.32 --------------
#14 17.32 
#14 17.32 *** Stopping MySQL
#14 17.32 + echo '*** Stopping MySQL'
#14 17.32 + kill 170
#14 17.33 + wait
#14 17.33 2021-12-14  9:45:54 0 [Note] /usr/bin/mysqld (initiated by: unknown): Normal shutdown
```
and got a clean result after removing the option `ROW_FORMAT=COMPRESSED`:
```
#14 22.96 + mysql --verbose '--user=mysql' '--protocol=socket' -uroot
#14 23.03 --------------
#14 23.14 DELETE FROM mysql.user
#14 23.14 --------------
#14 23.14 
#14 23.14 --------------
#14 23.14 DROP DATABASE IF EXISTS test
#14 23.14 --------------
#14 23.14 
#14 23.14 --------------
#14 23.14 CREATE DATABASE zipkin
#14 23.14 --------------
#14 23.14 
#14 23.14 --------------
#14 23.14 CREATE TABLE IF NOT EXISTS zipkin_spans (
#14 23.14   `trace_id_high` BIGINT NOT NULL DEFAULT 0 COMMENT 'If non zero, this means the trace uses 128 bit traceIds instead of 64 bit',
#14 23.14   `trace_id` BIGINT NOT NULL,
#14 23.14   `id` BIGINT NOT NULL,
#14 23.14   `name` VARCHAR(255) NOT NULL,
#14 23.14   `remote_service_name` VARCHAR(255),
#14 23.14   `parent_id` BIGINT,
#14 23.14   `debug` BIT(1),
#14 23.14   `start_ts` BIGINT COMMENT 'Span.timestamp(): epoch micros used for endTs query and to implement TTL',
#14 23.14   `duration` BIGINT COMMENT 'Span.duration(): micros used for minDuration and maxDuration query',
#14 23.14   PRIMARY KEY (`trace_id_high`, `trace_id`, `id`)
#14 23.14 ) ENGINE=InnoDB CHARACTER SET=utf8 COLLATE utf8_general_ci
#14 23.14 --------------
#14 23.14 
#14 23.14 --------------
#14 23.14 ALTER TABLE zipkin_spans ADD INDEX(`trace_id_high`, `trace_id`) COMMENT 'for getTracesByIds'
#14 23.14 --------------
#14 23.14 
#14 23.14 --------------
#14 23.14 ALTER TABLE zipkin_spans ADD INDEX(`name`) COMMENT 'for getTraces and getSpanNames'
#14 23.18 --------------
#14 23.18 
#14 23.18 --------------
#14 23.18 ALTER TABLE zipkin_spans ADD INDEX(`remote_service_name`) COMMENT 'for getTraces and getRemoteServiceNames'
#14 23.18 --------------
#14 23.18 
#14 23.18 --------------
#14 23.18 ALTER TABLE zipkin_spans ADD INDEX(`start_ts`) COMMENT 'for getTraces ordering and range'
#14 23.18 --------------
#14 23.18 
#14 23.18 --------------
#14 23.18 CREATE TABLE IF NOT EXISTS zipkin_annotations (
#14 23.18   `trace_id_high` BIGINT NOT NULL DEFAULT 0 COMMENT 'If non zero, this means the trace uses 128 bit traceIds instead of 64 bit',
#14 23.18   `trace_id` BIGINT NOT NULL COMMENT 'coincides with zipkin_spans.trace_id',
#14 23.18   `span_id` BIGINT NOT NULL COMMENT 'coincides with zipkin_spans.id',
#14 23.18   `a_key` VARCHAR(255) NOT NULL COMMENT 'BinaryAnnotation.key or Annotation.value if type == -1',
#14 23.18   `a_value` BLOB COMMENT 'BinaryAnnotation.value(), which must be smaller than 64KB',
#14 23.18   `a_type` INT NOT NULL COMMENT 'BinaryAnnotation.type() or -1 if Annotation',
#14 23.18   `a_timestamp` BIGINT COMMENT 'Used to implement TTL; Annotation.timestamp or zipkin_spans.timestamp',
#14 23.18   `endpoint_ipv4` INT COMMENT 'Null when Binary/Annotation.endpoint is null',
#14 23.18   `endpoint_ipv6` BINARY(16) COMMENT 'Null when Binary/Annotation.endpoint is null, or no IPv6 address',
#14 23.18   `endpoint_port` SMALLINT COMMENT 'Null when Binary/Annotation.endpoint is null',
#14 23.18   `endpoint_service_name` VARCHAR(255) COMMENT 'Null when Binary/Annotation.endpoint is null'
#14 23.18 ) ENGINE=InnoDB CHARACTER SET=utf8 COLLATE utf8_general_ci
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD UNIQUE KEY(`trace_id_high`, `trace_id`, `span_id`, `a_key`, `a_timestamp`) COMMENT 'Ignore insert on duplicate'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id_high`, `trace_id`, `span_id`) COMMENT 'for joining with zipkin_spans'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id_high`, `trace_id`) COMMENT 'for getTraces/ByIds'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`endpoint_service_name`) COMMENT 'for getTraces and getServiceNames'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`a_type`) COMMENT 'for getTraces and autocomplete values'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`a_key`) COMMENT 'for getTraces and autocomplete values'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.25 ALTER TABLE zipkin_annotations ADD INDEX(`trace_id`, `span_id`, `a_key`) COMMENT 'for dependencies job'
#14 23.25 --------------
#14 23.25 
#14 23.25 --------------
#14 23.28 CREATE TABLE IF NOT EXISTS zipkin_dependencies (
#14 23.28   `day` DATE NOT NULL,
#14 23.28   `parent` VARCHAR(255) NOT NULL,
#14 23.28   `child` VARCHAR(255) NOT NULL,
#14 23.28   `call_count` BIGINT,
#14 23.28   `error_count` BIGINT,
#14 23.28   PRIMARY KEY (`day`, `parent`, `child`)
#14 23.28 ) ENGINE=InnoDB CHARACTER SET=utf8 COLLATE utf8_general_ci
#14 23.28 --------------
#14 23.28 
#14 23.28 --------------
#14 23.28 GRANT ALL PRIVILEGES ON zipkin.* TO zipkin@'%' IDENTIFIED BY 'zipkin' WITH GRANT OPTION
#14 23.28 --------------
#14 23.28 
#14 23.28 --------------
#14 23.28 FLUSH PRIVILEGES
#14 23.28 --------------
#14 23.28 
#14 23.28 + echo '*** Stopping MySQL'
#14 23.28 *** Stopping MySQL
#14 23.28 + kill 172
#14 23.28 + wait
#14 23.28 2021-12-14 10:11:46 0 [Note] /usr/bin/mysqld (initiated by: unknown): Normal shutdown
```

Interestingly, the build had been failing when I firstly made this PR, which does not ATM with the reported error messages. Plus, the same error messages are also found for `amd64` and `arm64`. I am not sure if this would make any influence to any aspects of the product.

Reply:
I just tried with a different PR. I think we don't need this anymore: https://github.com/openzipkin/zipkin/pull/3404

Reply:
Please cherry-pick `5cad83f942c6be5e388f06aab463ea2525f197ef` (see https://github.com/openzipkin/zipkin/pull/3404/commits/5cad83f942c6be5e388f06aab463ea2525f197ef) into this PR and I will merge it.

Reply:
Done ^^ thanks for testing this.